plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/RetryTemplateService.java

```diff
@@ -0,0 +1,168 @@
+                        KeystoneListProjectsResponse.class);
+        if (Objects.nonNull(response.getBody())) {
+            if (!CollectionUtils.isEmpty(response.getBody().getProjects())) {
+                return response.getBody().getProjects().get(0);
+            }
+        }
+        throw new RestClientException("query project info result is null.");
+    }
+
+
+    /**
+     * Query metric data using RestTemplate with Spring-Retry.
+     *
+     * @param httpRequestBase httpRequestBase
+     * @return ShowMetricDataResponse
+     */
+    @Retryable(retryFor = RestClientException.class, maxAttempts = RETRY_TIMES,
+            backoff = @Backoff(delay = DELAY_MILLISECONDS, multiplier = DELAY_MULTIPLIER))
+    public ShowMetricDataResponse queryMetricData(HttpRequestBase httpRequestBase)
+            throws RestClientException {
+        HttpEntity<String> entity = new HttpEntity<>("parameters", getHttpHeaders(httpRequestBase));
+        ResponseEntity<ShowMetricDataResponse> responseEntity =
+                restTemplate.exchange(httpRequestBase.getURI(), HttpMethod.GET, entity,
+                        ShowMetricDataResponse.class);
+        if (responseEntity.getStatusCode().equals(HttpStatus.OK)) {
+            return responseEntity.getBody();
+        }
+        throw new RestClientException("query metric data failed.");
+    }
+
+    /**
+     * Query metric item list using RestTemplate with Spring-Retry.
+     *
+     * @param httpRequestBase httpRequestBase
+     * @return ShowMetricDataResponse
+     */
+    @Retryable(retryFor = RestClientException.class, maxAttempts = RETRY_TIMES,
+            backoff = @Backoff(delay = DELAY_MILLISECONDS, multiplier = DELAY_MULTIPLIER))
+    public ListMetricsResponse queryMetricItemList(HttpRequestBase httpRequestBase) {
+        HttpEntity<String> entity = new HttpEntity<>("parameters", getHttpHeaders(httpRequestBase));
+        ResponseEntity<ListMetricsResponse> responseEntity =
+                restTemplate.exchange(httpRequestBase.getURI(), HttpMethod.GET, entity,
+                        ListMetricsResponse.class);
+        if (responseEntity.getStatusCode().equals(HttpStatus.OK)) {
+            return responseEntity.getBody();
+        }
+        throw new RestClientException("query metric item list failed.");
+    }
+
+    /**
+     * Batch query metric data using RestTemplate with Spring-Retry.
+     *
+     * @param httpRequestBase httpRequestBase
+     * @param requestBody     requestBody
+     * @return BatchListMetricDataResponse
+     */
+    @Retryable(retryFor = RestClientException.class, maxAttempts = RETRY_TIMES,
+            backoff = @Backoff(delay = DELAY_MILLISECONDS, multiplier = DELAY_MULTIPLIER))
+    public BatchListMetricDataResponse batchQueryMetricData(HttpRequestBase httpRequestBase,
+                                                            String requestBody) {
+        HttpEntity<String> entity = new HttpEntity<>(requestBody,
```

Comment:
everywhere -> can we call it requestEntity instead of just entity? so that is is more clear.

Reply:
Ok.